ssr/src/page/post_view/bet.rs

```diff
@@ -348,33 +374,58 @@ fn HNAwaitingResults(
     };
 
     view! {
-        <div class="flex flex-col w-full items-center gap-1 p-4 shadow-sm">
-            <div class="flex flex-row w-full justify-center items-end gap-4">
+        <div class="flex flex-col gap-1 items-center p-4 w-full shadow-sm">
+            <div class="flex flex-row gap-4 justify-center items-end w-full">
                 <div class="relative flex-shrink-0 drop-shadow-lg">
                     <Icon class="w-12 h-12 md:w-14 md:h-14 lg:w-16 lg:h-16" icon=hn_icon />
                     <CoinStateView
-                        class="absolute bottom-0 -right-3 h-7 w-7 md:w-9 md:h-9 lg:w-11 lg:h-11"
+                        class="absolute bottom-0 -right-3 w-7 h-7 md:w-9 md:h-9 lg:w-11 lg:h-11"
                         coin
                     />
                 </div>
+
                 <div class="w-1/2 md:w-1/3 lg:w-1/4">
                     <BetTimer post refetch_bet participation />
                 </div>
             </div>
-            <p class="text-center text-white bg-black/15 rounded-full p-1 ps-2">
-                You staked {bet_amount}tokens on {bet_direction_text}Result is still pending
+            <p class="p-1 text-center text-white rounded-full bg-black/15 ps-2">
+                You staked {bet_amount} tokens on {bet_direction_text} Result is still pending
             </p>
         </div>
     }
 }
-
 #[component]
 pub fn HNUserParticipation(
     post: PostDetails,
     participation: BetDetails,
     refetch_bet: Trigger,
 ) -> impl IntoView {
+    let place_bet_action = create_action(
+        move |(canisters, bet_direction, bet_amount): &(Canisters<true>, BetKind, u64)| {
+            let post_can_id = post.canister_id;
+            let post_id = post.post_id;
+            let cans = canisters.clone();
+            let bet_amount = *bet_amount;
+            let bet_direction = *bet_direction;
+            async move {
+                match bet_on_post(cans, bet_amount, bet_direction, post_id, post_can_id).await {
+                    Ok(_) => Some(()),
+                    Err(e) => {
+                        log::error!("{e}");
+                        None
+                    }
+                }
+            }
+        },
+    );
+    let place_bet_res = place_bet_action.value();
+    create_effect(move |_| {
+        if place_bet_res().flatten().is_some() {
+            refetch_bet.notify();
+        }
+    });
```

Comment:
You can call refetch_bet.notify inside the action itself

Reply:
done

Reply:
should i merge this branch with updated main and resolve conflicts before pushing here @rupansh-gob